instrumentation/spring-rabbit/src/main/java/brave/spring/rabbit/TracingRabbitListenerAdvice.java

```diff
@@ -67,16 +67,19 @@
     TraceContextOrSamplingFlags extracted = extractTraceContextAndRemoveHeaders(message);
 
     // named for BlockingQueueConsumer.nextMessage, which we can't currently see
-    Span consumerSpan = tracer.nextSpan(extracted).kind(CONSUMER).name("next-message").start();
+    Span consumerSpan = tracer.nextSpan(extracted).kind(CONSUMER).name("next-message");
+    Span listenerSpan = tracer.newChild(consumerSpan.context()).name("on-message");
```

Comment:
@Logic-32 the bug was yeah a race.. the underlying recorder loses the clock for the parent on finish, so I create an unstarted span here to fix the race. It is a workaround really

Reply:
Would it be worthwhile to have a SpanMap like equivalent for TraceContext which simply tracks the clock?  Would you be able to guarantee the same weakness of the Map while still maintaining consistency between arbitrarily created TraceContexts?